src/Validator.ts

```diff
@@ -8,7 +8,7 @@ export class Validator {
     ) {
     }
 
-    validate<T>(target: T, options?: ValidationOptions): ValidationResult<T> {
+    validate<T extends { constructor: Function }>(target: T, options?: ValidationOptions): ValidationResult<T> {
```

Comment:
Had to add this, otherwise line further down didn't know about the `constructor` property.